contracts/ics-escrow/src/contract.rs

```diff
@@ -0,0 +1,138 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    to_binary, Binary, Deps, DepsMut, Empty, Env, MessageInfo, Response, StdResult, WasmMsg,
+};
+use cw2::set_contract_version;
+
+use crate::error::ContractError;
+use crate::msg::{ExecuteMsg, InstantiateMsg, QueryMsg};
+use crate::state::{ADMIN_ADDRESS, CHANNEL_ID};
+
+const CONTRACT_NAME: &str = "crates.io:ics-escrow";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    let admin_address = deps.api.addr_validate(&msg.admin_address)?;
+    ADMIN_ADDRESS.save(deps.storage, &admin_address)?;
+    CHANNEL_ID.save(deps.storage, &msg.channel_id)?;
+    Ok(Response::new()
+        .add_attribute("action", "instantiate")
+        .add_attribute("admin_address", admin_address.to_string())
+        .add_attribute("channel_id", msg.channel_id))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::Withdraw {
```

Comment:
Isn't there supposed to be some kind of deposit NFT function for escrow?

Reply:
The spec doesn't say anything about the interface for escrows so we keep it light. NFTs get deposited in the escrow once the contract has sent an IBC packet sending them away in the `ReceiveNft` method.

Reply:
Channel escrow deposit happens here: https://github.com/public-awesome/ics721/blob/c1d19347ba00aebee72f5c64b6a9fe5a066b93ac/contracts/cw-ics721-bridge/src/contract.rs#L353-L361

Reply:
@ezekiiel okay so you're depositing via a TransferNFT message that transfers the NFT to the escrow right? 
Shouldn't we then have a ExecuteMsg::TransferNFT block in escrow then?

Reply:
The transfer method gets called on the cw721 contract. [Here](https://github.com/CosmWasm/cw-nfts/blob/342120f572ff1e3368b014c4bfdc83f25b664666/contracts/cw721-base/src/msg.rs#L26-L27) is the message in question. The escrow actually doesn't even need to know that it's received the NFT. We're just doing this for bookkeeping.

Reply:
@ezekiiel what do you mean by "the cw721 contract" do you mean 
1) escrow contract extends/implements cw721 implicitly
2) NFT is actually being stored somewhere else other than this file

If #1 noob question can you show how this is done?

Reply:
> @humanalgorithm: what do you mean by "the cw721 contract"

I just mean whatever cw721 was instantiated for a the particular ICS 721 NFT being stored in the escrow. We don't need to add a deposit method to the escrow because we can just call the `Transfer` method on that cw721 contract to move it into the escrow.

The reason we need a withdraw method and not a deposit one is that only the owner of the NFT can do a transfer. If we want to withdraw the escrow needs to execute the message as it is the owner of NFTs inside of it. For depositing, the bridge can execute the method.

Does this make sense? I may be misunderstanding your question here.